code/service/src/main/scala/com/namely/chiefofstate/CosSnapshotStore.scala

```diff
@@ -0,0 +1,117 @@
+ */
+class CosSnapshotStore(config: Config) extends SnapshotStore {
+
+  implicit val ec: ExecutionContext = context.dispatcher
+  implicit val system: ActorSystem = context.system
+
+  private[chiefofstate] val readJournal: JdbcReadJournal = PersistenceQuery(system)
+    .readJournalFor[JdbcReadJournal](JdbcReadJournal.Identifier)
+
+  /**
+   * implements loadAsync for akka persistence snapshots to read from the
+   * jdbc journal and return an empty "snapshot" pointing to the highest
+   * persisted sequence number. This works since COS persists each resulting
+   * state in the journal.
+   *
+   * @param persistenceId persistence ID to restore
+   * @param criteria snapshot criteria to restore
+   * @return optional selected snapshot
+   */
+  override def loadAsync(persistenceId: String,
+                         criteria: SnapshotSelectionCriteria
+  ): Future[Option[SelectedSnapshot]] = {
+    logger.debug(s"loadAsync, persistenceId=$persistenceId, criteria=$criteria")
+
+    // get the events for this entity
+    val source: Source[EventEnvelope, NotUsed] = readJournal
+      .currentEventsByPersistenceId(persistenceId, criteria.minSequenceNr, criteria.maxSequenceNr)
+
+    val maxEnvelopeFuture: Future[Option[EventEnvelope]] = source
+      // fold to the max sequence number
+      .runFold[Option[EventEnvelope]](None)((maxEnvelope: Option[EventEnvelope], nextEnvelope: EventEnvelope) => {
+        maxEnvelope.map(prior => {
+          if (prior.sequenceNr < nextEnvelope.sequenceNr) nextEnvelope else prior
+        })
+      })
+
+    maxEnvelopeFuture
+      .map(_.map((maxEnvelope: EventEnvelope) => {
+        // FIXME, make a debug log
+        logger.info(s"restoring, persistenceId=${maxEnvelope.persistenceId}, sequenceNo=${maxEnvelope.persistenceId}")
+
+        SelectedSnapshot(
+          metadata = SnapshotMetadata(
+            persistenceId = maxEnvelope.persistenceId,
+            sequenceNr = maxEnvelope.sequenceNr,
+            timestamp = maxEnvelope.timestamp
+          ),
+          StateWrapper.defaultInstance
+        )
+      }))
+  }
+
+  /**
+   * dummy implementation of the async save of snapshots, which does nothing
+   *
+   * @param metadata ignored
+   * @param snapshot ignored
+   * @return successful future
+   */
+  override def saveAsync(metadata: SnapshotMetadata, snapshot: Any): Future[Unit] = {
+    Future.unit
+  }
+
+  /**
+   * dummy deletion method, since there are no snapshots in the db
+   *
+   * @param metadata ignored
+   * @return successful future
+   */
+  override def deleteAsync(metadata: SnapshotMetadata): Future[Unit] = {
+    Future.unit
+  }
+
+  /**
+   * dummy implementation of deletion, since there are no snapshots in the db
+   *
+   * @param persistenceId ignored
+   * @param criteria ignored
+   * @return successful future
+   */
+  override def deleteAsync(persistenceId: String, criteria: SnapshotSelectionCriteria): Future[Unit] = {
+    Future.unit
```

Comment:
```suggestion
    Future.successful(())
```

Reply:
They are synonymous, and I actually prefer this syntax to the empty success. Thoughts?